[MRG] website navigation improvements

aabadie opened the pull request:
This should fix #1071.

I took as example the layout and stylesheet from [scipy-lecture-notes](https://github.com/scipy-lectures/scipy-lecture-notes).
The css changeset also contains a fix for the code example overlapping the sidebar. Thus I'll close #1078.

GaelVaroquaux commented on doc/conf.py:
```diff
@@ -31,7 +31,7 @@
 # extensions coming with Sphinx (named 'sphinx.ext.*') or your custom ones.
 extensions = ['sphinx.ext.autodoc', 'sphinx.ext.autosummary',
-              'sphinx.ext.pngmath', 'sphinx.ext.intersphinx',
+              'sphinx.ext.imgmath', 'sphinx.ext.intersphinx',
```
We probably need to merge your other PR before we merge that one.

GaelVaroquaux commented:
+1 in general. This is nice!

Bonus point if you can fix the bug that's also present in scipy-lectures where the subsection don't get bold.

aabadie commented:
It should be ok now. Main changes:
* in the table of content sidebar section are expanded while scrolling and the text of the current section is now bold (before it was a rectangle background)
* the same logic applied to the examples : current examples section text displayed in bold
* I fixed the positioning of the TOC while scrolling
* the top navigation bar has a sticky behaviour : when on scroll it stays at the top of the window. The bottom navigation was useless so I removed it.
* I backported the css for auto wrapping python highlights from scipy-lectures-notes.

> Bonus point if you can fix the bug that's also present in scipy-lectures where the subsection don't get bold.

I had a look but couldn't find a solution to this problem. To me the issue comes from [here](https://github.com/nilearn/nilearn/blob/master/doc/themes/nilearn/layout.html#L109) : the subsection are still hidden when activating their parent section, preventing the subsection to become `active` and thus `bold`.

bthirion commented:
Overall, this looks good, but personally, I don't like the interaction with the right panel that comes and disappears when you move the mouse. Actually, you lose it when you scroll through the page.
Maybe I'm a bit old-fashioned...

aabadie commented:
> I don't like the interaction with the right panel that comes and disappears when you move the mouse. Actually, you lose it when you scroll through the page.

What you describe is weird. I don't have this but only tested on Linux with Firefox : The horizontal navigation bar sticks to the top while scrolling and the right side bar as well. In the right side bar, the current section is highlighted with bold text.

I gave a try to chromium and notice that the horizontal bar doesn't take the full with of the page, this is a bug. The second problem is that something it has a jumpy effect when scrolling with the mouse wheel.

aabadie commented:
Small update on this one: the problem with chrome was due to the "sticky" property I used in the css, which is only available in firefox...

I pushed a change that should be ok now (only tested with chromium for the moment). Maybe someone can also test with Safari and Chrome using MacOS or Windows ?

bthirion commented:
I still have the same issue with Chrome and firefox.

aabadie commented:
@bthirion, I can't reproduce your issue.
Have you tried the [latest build on circle ci](https://circle-artifacts.com/gh/aabadie/nilearn/341/artifacts/0/home/ubuntu/nilearn/doc/_build/html/index.html) ?

What version of Chrome/Firefox are you using ? (I'm running Firefox 46.0 and Chromium 49.0)

bthirion commented:
On 02/05/2016 15:11, Alexandre Abadie wrote:
>
> @bthirion <https://github.com/bthirion>, I can't reproduce your issue.
> Have you tried the latest build on circle ci 
> <https://circle-artifacts.com/gh/aabadie/nilearn/341/artifacts/0/home/ubuntu/nilearn/doc/_build/html/index.html> 
> ?
>
> What version of Chrome/Firefox are you using ? (I'm running Firefox 
> 46.0 and Chromium 49.0)
>
Mozilla Firefox 45.0.2
Chromium 49.0.2623.108 Ubuntu 14.04

Maybe there is a misunderstanding on what I should see :-)

> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub 
> <https://github.com/nilearn/nilearn/pull/1082#issuecomment-216230746>
>

aabadie commented:
> Maybe there is a misunderstanding on what I should see :-)

I don't think so. I could trigger a spurious behaviour on Mac OS using both Firefox and Chrome which seems to be similar to what you described initially : the sidebar on the right disappears while scrolling. To me, this is a css issue. I'll try to fix it.

bthirion commented:
On 04/05/2016 09:36, Alexandre Abadie wrote:
>
>     Maybe there is a misunderstanding on what I should see :-)
>
> I don't think so. I could trigger a spurious behaviour on Mac OS using 
> both Firefox and Chrome which seems to be similar to what you 
> described initially : the sidebar on the right disappears while 
> scrolling. To me, this is a css issue. I'll try to fix it.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub 
> <https://github.com/nilearn/nilearn/pull/1082#issuecomment-216767187>
>
I'm at AT. I can come and discuss with you within 1h.

B

GaelVaroquaux commented:
@aabadie : before we merge this, we need an media css selector so that the topbar is not sticky when the screen is small: it steals to much screen real estate. This is important as we use the website for teaching (and so people look at it on mobile phones).

GaelVaroquaux commented:
Also the side menu doesn't look right when it is collapsed and is displayed by a hover.

aabadie commented:
Revamping this one (after such a long time).

I finally rebased this PR, reworked a bit the website layout and dropped the initial css changes. This PR provides sticky top menu bar and side menu while scrolling.

I also fixed a bug: when the page is refreshed while reading the user guide, it automatically scrolls to the previous position, which is great. But the side menu disappear because its position is not restored, one has to scroll a bit to get it back. This is the bug that is now also fixed by this PR.

I also checked with small screens (I tried with zooming a lot and resizing the window): the top menu is not sticky in this case so there will be no problem for you when you teach.

I tested on Firefox and Chrome (Linux only) and everything looks fine AFAICS.

> Also the side menu doesn't look right when it is collapsed and is displayed by a hover.

I cannot display it when it is collapsed, even on the online website. How/where this can be done ?

GaelVaroquaux commented:
@aabadie : it's great that you are doing this. Thanks! We are prepare a release of nilearn soon, and there has been a lot of exciting activity on it. I would be excited if we could have a rewamp of the website.

GaelVaroquaux commented:
I've tried it on my computer, and it looks great. There is one regression compared to the site that is on line: when the screen is small, or the fonts are huge, in the new version, there is a blue band on the side, that is not present in the old version.

aabadie commented:
> there is a blue band on the side, that is not present in the old version.

indeed, there was an issue with the width and height of the main menu. It is fixed now.

GaelVaroquaux commented:
For the curious, the results are visible here: https://352-41858666-gh.circle-artifacts.com/0/home/circleci/project/doc/_build/html/index.html

GaelVaroquaux commented:
I'm sorry, @aabadie : it's still not looking right with the right side: the page can be shifted to the left to reveal a blue band.

GaelVaroquaux commented:
OK, the problem is now fixed on a local build on my computer. I'll wait for CI to finish, but I have good hopes that this can be merge. Thanks a lot!

GaelVaroquaux commented:
For perusal, the rendered docs are here: https://355-41858666-gh.circle-artifacts.com/0/home/circleci/project/doc/_build/html/introduction.html#python-for-neuroimaging-a-quick-start

aabadie commented:
> Still broken on mobile phone :(

By this PR or is it already the case in master ? (I can't verify, my phone is broken...)

GaelVaroquaux commented:
> By this PR or is it already the case in master ?

Good shot: only a small number of pages are broken, and it has nothing to
do with this PR (it's related to lines too long).

Merging!! Thanks!
